features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -350,11 +348,23 @@ void ATHandler::fill_buffer()
        ssize_t len = _fileHandle->read(_recv_buff + _recv_len, sizeof(_recv_buff) - _recv_len);
         if (len > 0) {
             _recv_len += len;
-           at_debug("\n----------readable----------: %d\n", _recv_len);
-           for (size_t i = _recv_pos; i < _recv_len; i++) {
-               at_debug("%c", _recv_buff[i]);
+
+#if MBED_CONF_MBED_TRACE_ENABLE // AT debug printing
+           char *p = _recv_buff + _recv_len - len;
+           for (ssize_t i = 0; i < len; i++) {
```

Comment:
how is this loop different from one on the line 1055 ? They look a like (should it be own small helper that would be called from these 2 places) ?